suite/tests/client-interface/stolen-reg-index.c

```diff
@@ -0,0 +1,92 @@
+/* **********************************************************
+ * Copyright (c) 2022 Arm Limited   All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* Executes two indexed memory instructions, one using X28 for index register,
+ * the other W28. Intention is for DR to call drutil_insert_get_mem_addr() in
+ * order to test the 'if (index == stolen)' clause in
+ * drutil_insert_get_mem_addr_arm() in the case of W28.
+ */
+
+#ifndef ASM_CODE_ONLY /* C code */
+
+#    include "tools.h"
+
+/* In asm code. */
+void
+indexed_mem_test(int *val);
+
+int
+main(int argc, char *argv[])
+{
+    int value = 41;
+    indexed_mem_test(&value);
+    if (value != 42) {
+        fprintf(stderr, "indexed_mem_test() failed with %d, expected 42.\n", value);
```

Comment:
Our convention is to always use print() in tests which goes to stderr to interleave reliably with DR's stderr output, and is flushed everywhere (incl Windows cygwin).  Ditto below.

Reply:
Done.